sympy/core/equation.py

```diff
@@ -0,0 +1,359 @@
+        """
+        Swaps the lhs and the rhs.
+        """
+        return Equation(self.rhs, self.lhs)
+    
+    @property
+    def free_symbols(self):
+        ret =self.lhs.free_symbols 
+        ret.update(self.rhs.free_symbols)
+        return ret
+
+    def _applyfunc(self, func, *args, **kwargs):
+        # Assume if the expression has an attribute of name `func` that should override any general function
+        # Because there are name conflicts (e.g. `transpose` and `Matrix.transpose`) if either rhs or lhs
+        # has the attribute we will try to apply it to both. This will raise an error if both sides do
+        # not support the operation.
+
+        side=kwargs.pop('Eqn_apply_side',None)
+        if (side=='both'):
+            if (hasattr(self.lhs,str(func))) or (hasattr(self.rhs,str(func))):
+                return Equation(getattr(self.lhs,str(func))(*args, **kwargs),getattr(self.rhs,str(func))(*args, **kwargs))
+            else:
+                return Equation(func(self.lhs, *args, **kwargs), func(self.rhs, *args, **kwargs))
+        elif (side == 'lhs'):
+            if (hasattr(self.lhs,str(func))):
+                return Equation(getattr(self.lhs,str(func))(*args, **kwargs),self.rhs)
+            else:
+                return Equation(func(self.lhs, *args, **kwargs), self.rhs)
+        elif (side == 'rhs'):
+            if (hasattr(self.rhs,str(func))):
+                return Equation(self.lhs, getattr(self.rhs,str(func))(*args, **kwargs))
+            else:
+                return Equation(self.lhs, func(self.rhs, *args, **kwargs))
+        else:
+            raise ValueError('keyword `Eqn_apply_side` must be one of "both", "lhs" or "rhs".')
+
+    def applyfunc(self, func, *args, **kwargs):
+        """
+        If either side of the equation has a defined subfunction (attribute) of name ``func``, that will be applied
+        instead of the global function. The operation is applied to both sides.
+        """
+        return self._applyfunc(func, *args, **kwargs, Eqn_apply_side='both')
+    
+    def applylhs(self, func, *args, **kwargs):
+        """
+        If lhs side of the equation has a defined subfunction (attribute) of name ``func``, that will be applied
+        instead of the global function. The operation is applied to only the lhs.
+        """
+        return self._applyfunc(func, *args, **kwargs, Eqn_apply_side='lhs')
+
+    def applyrhs(self, func, *args, **kwargs):
+        """
+        If rhs side of the equation has a defined subfunction (attribute) of name ``func``, that will be applied
+        instead of the global function. The operation is applied to only the rhs.
+        """
+        return self._applyfunc(func, *args, **kwargs, Eqn_apply_side='rhs')
+
+#####
+# Overrides of binary math operations
+#####
+
+    _op_priority = 11.0 # This makes sure the rules for equations are applied before those for expressions
+                        # which have _op_priority = 10.0
```

Comment:
Is `_op_priority` needed?

Can you give an example of why this is needed?

In binary operations Expr checks if the other arg sympifies to Expr and returns NotImplemented otherwise so I would have thought that the `_op_priority` doesn't do anything for a non-Expr class.

Reply:
I will have to generate the issue again. It related to failure of Symbol*Equation versus Equation*Symbol. More later.

Reply:
Since changing to the `_binary_op` method of handling the arithmetic I cannot reproduce the problem. I am taking this out.